### PR TITLE
SQL: Ibis frontend refactor

### PR DESCRIPTION
**Pull request by webmiche**
This patch refactors the ibis frontend to use more of multipledispatch.

### CODE REVIEW

File: experimental/sql/src/ibis_frontend.py [hunk 1]
```diff
@@ -36,7 +30,7 @@ def convert_datatype(type_: ibis.expr.datatypes) -> id.DataType:
   raise KeyError(f"Unknown datatype: {type(type_)}")
 
 
-def visit_schema(schema: ibis.expr.schema.Schema) -> Region:  #type: ignore
+def visit_schema(schema: ibis.expr.schema.Schema) -> Region:
```
Comment (ingomueller-net) [inline experimental/sql/src/ibis_frontend.py:33]: Why does this function and the next not use multipledispatch? Add a small comment?

File: experimental/sql/src/ibis_frontend.py [hunk 2]
```diff
@@ -50,92 +44,63 @@ def visit_ibis_expr_list(l: List[ibis.expr.types.Expr]) -> Region:
...
-    table = Region.from_operation_list([visit(op.table)])
-    metrics = visit_ibis_expr_list(op.metrics)
-    new_op = id.Aggregation.get(table, metrics)
-    return new_op
-  raise KeyError(f"Unknown tableExpr: {type(op)}")
-
-
-@dispatch(ibis.expr.types.StringColumn)
+    op: ibis.expr.types.Expr) -> Operation:
+  return visit(op.op())
```
Comment (ingomueller-net) [inline experimental/sql/src/ibis_frontend.py:50]: This is nice!

**Review by ingomueller-net — APPROVED** [review]
Cool, looks much better, IMO!

File: experimental/sql/src/ibis_frontend.py (hunk 1 above)
Comment (webmiche) [reply experimental/sql/src/ibis_frontend.py:35]: They return Regions as they work on multiple nodes in the ibis tree whereas the other functions return Operations for the single tree nodes they work. Added a comment :)

**Pull request merged**